src/main/kotlin/org/opensearch/replication/task/shard/ShardReplicationTask.kt

```diff
@@ -222,23 +224,32 @@ class ShardReplicationTask(id: Long, type: String, action: String, description:
                         logDebug("pushed to sequencer $fromSeqNo-$toSeqNo")
                         changeTracker.updateBatchFetched(true, fromSeqNo, toSeqNo, changesResponse.changes.lastOrNull()?.seqNo() ?: fromSeqNo - 1,
                             changesResponse.lastSyncedGlobalCheckpoint)
+
+                        followerClusterStats.stats[followerShardId]!!.remoteCheckpoint = changesResponse.lastSyncedGlobalCheckpoint
+                        followerClusterStats.stats[followerShardId]!!.opsRead.addAndGet(changesResponse.changes.size.toLong())
+
                     } catch (e: OpenSearchTimeoutException) {
                         logInfo("Timed out waiting for new changes. Current seqNo: $fromSeqNo. $e")
                         changeTracker.updateBatchFetched(false, fromSeqNo, toSeqNo, fromSeqNo - 1,-1)
                     } catch (e: NodeNotConnectedException) {
+                        followerClusterStats.stats[followerShardId]!!.opsReadFailures.addAndGet(1)
                         logInfo("Node not connected. Retrying request using a different node. ${e.stackTraceToString()}")
                         delay(backOffForNodeDiscovery)
                         changeTracker.updateBatchFetched(false, fromSeqNo, toSeqNo, fromSeqNo - 1,-1)
                     } catch (e: Exception) {
+                        followerClusterStats.stats[followerShardId]!!.opsReadFailures.addAndGet(1)
```

Comment:
opsReadFailures.add instead of opsReadFailures.addAndGet where ever we are not reading the value?

Reply:
There is no `add` function available.

Reply:
There is no `add` function available.